components/http_client.rst

```diff
@@ -583,7 +611,16 @@ catching ``TransportExceptionInterface`` in the foreach loop::
 
     foreach ($client->stream($responses) as $response => $chunk) {
         try {
-            if ($chunk->isLast()) {
+            if ($chunk->isTimeout()) {
+                // ... decide what to do when a timeout occurs
+                // if you want to stop a response that timed out, don't miss
+                // calling $response->cancel() or the destructor of the response
+                // will try to complete it one more time
+            } elseif ($chunk->isFirst()) {
+                // if you want to check the status code, you must do it when the
+                // first chunk arrived, using $response->getStatusCode();
+                // not doing so might trigger an HttpExceptionInterface
```

Comment:
Because the response headers might not be arrived yet? I am not sure to understand in which case it can happen.

Reply:
It happens because that's how the first chunk behaves : if status code is unchecked after it, an exception is thrown. I'm going to improve this

Reply:
It was a little bit confusing that the exception is thrown from the iterator(generator), so you can not catch it inside a for-each cycle and continue to process other requests.